애니 Stub 및 Test 추가

jaycobcoder opened the pull request:
## 📝 개요
애니 Stub 및 Test

## 🚀 변경사항
- `AttractionElement` 중 `ART` -> `DRAWING` 변경,
- `AnimeReq` 및 `Anime` 생성 메소드 제거,
- `AnimeController` 추가,
- `AnimeService` 및 `AnimeServiceStub` 추가,
- `Anime` 테스트 추가

## 🔗 관련 이슈
#6

jaycobcoder committed:
refactor: enum AttractionElement ART 수정 #6

클라이언트가 작화를 입덕 포인트로 선택했을 때 ART에서 DRAWING으로 변경하였습니다.

jaycobcoder committed:
cleanup: AnimeReq 삭제 #6

애니 등록은 관리자에서 이루어질 것으로 예상합니다. 따라서 애니 등록 DTO를 Anime 패키지에서 삭제하였습니다.

jaycobcoder committed:
feat: Anime Stub 추가 #6

jaycobcoder committed:
test: Anime Controller, Anime Service 테스트 추가 및 문서화 #6

hanyMK reviewed (approved):
👍

FaberJoo commented:
<img width="870" alt="image" src="https://github.com/oduck-team/oduck-server/assets/79781818/f555dfe5-482b-4834-a21b-e9fa4bd1e61f">

다음과 같은 이유로 테스트 통과가 안 되네요.

restDocs responseField에 animeId가 빠진 것 같습니다.

jaycobcoder committed:
test: restDocs의 responseField에 animeId 추가 #6

jaycobcoder committed:
refactor: AnimeRes의 내부 클래스 BroadCast -> Broadcast로 변경 #6

jaycobcoder commented:
> <img alt="image" width="870" src="https://user-images.githubusercontent.com/79781818/272161238-f555dfe5-482b-4834-a21b-e9fa4bd1e61f.png">
> 다음과 같은 이유로 테스트 통과가 안 되네요.
> 
> restDocs responseField에 animeId가 빠진 것 같습니다.

헉..! 컨트롤러 테스트 통과하고, 서비스 테스트 돌릴 때 응답 dto에 anime_id도 응답해주는 게 좋겠다고 생각해서 dto에 애니 고유 식별자를 추가했었습니다. 서비스는 테스트 통과해서 넘어갔는데 컨트롤러를 테스트 돌려보지 못했네요😭 수정했습니다!

FaberJoo commented:
api문서(src/docs/asciidoc/index.adoc)도 작성해 보시면 좋을 것 같습니다.

jaycobcoder committed:
refactor: Anime Entity 연관 관계 수정 #6

Anime와 관련된 AnimeOriginalAuthor, AnimeStudios, AnimeVoiceActors, AnimeGenres는 batchsize 설정으로 한 번에 가져오고, Bookmark, ShortReview, StarRating, AttractionPoint는 연관 관계를 제거하였습니다.

jaycobcoder committed:
refactor: Anime 엔티티 bookmark count 추가 #6

jaycobcoder committed:
refactor: 요구사항에 따라 애니 값만 리턴 #6

jaycobcoder committed:
test: 요구사항에 따라 애니 값만 리턴, 설명 수정 #6

jaycobcoder committed:
docs: index.adoc의 anime 도메인 추가 #6

FaberJoo commented on src/test/java/io/oduck/api/e2e/anime/AnimeControllerTest.java:
```diff
@@ -0,0 +1,145 @@
+            );
+
+            //then
+            actions
```
![image](https://github.com/oduck-team/oduck-server/assets/79781818/7794faf1-5f45-4f6d-b78e-fb4113855f99)
![image](https://github.com/oduck-team/oduck-server/assets/79781818/8d03d09c-b553-4692-b928-61264555d0f0)
![image](https://github.com/oduck-team/oduck-server/assets/79781818/1bf96958-8389-4e11-a735-c317a16a8cb4)


테스트 검증 및 문서화에는 응답 객체에 score가 없는데,
응답 바디 `AnimeRes` 객체에 score 객체가 남아있어서 위와 같은 문제가 발생하네요.

jaycobcoder committed:
refactor: 요구사항에 따라 애니 값만 리턴 #

FaberJoo reviewed (approved):
🚀
